File: lesson5/HW/service/UserService.java
package lesson5.HW.service;

import java.util.List;

import lesson5.HW.model.User;

public interface UserService {
	void saveUser(User user);

	void deleteUser(User user);

	void updateUser(int index, User user);

	List<User> getAllUsers();

	void addFunds(int index, float funds);

	void removeFunds(int index, float funds);
}

File: Seminar7/DZ/py/UserService.py
from abc import ABC, abstractmethod


class UserService:
    @abstractmethod
    def saveUser(self, user):
        pass

    @abstractmethod
    def deleteUser(self, user):
        pass

    @abstractmethod
    def updateUser(self, index, user):
        pass

    @abstractmethod
    def getAllUsers(self):
        pass

    @abstractmethod
    def addFunds(self, index, funds):
        pass

    @abstractmethod
    def removeFunds(self, index, funds):
        pass
